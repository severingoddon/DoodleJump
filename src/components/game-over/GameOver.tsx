import React from 'react';
import './game-over-css.css';

let scoreClassName = 'result-score';

export const setThemeGameOverScore = (isDark: boolean) =>{
    if(isDark) scoreClassName = 'result-score-dark';
    else scoreClassName = 'result-score';
};

export default (props: any) => {

    const style = {};

    return(
        <div className="game-over-screen" style={style}>
            <div className="game-over">Game over!</div>
            <div data-testid="finalscore" className={scoreClassName}>Your score is: {props.score}</div>
        </div>
    );
};