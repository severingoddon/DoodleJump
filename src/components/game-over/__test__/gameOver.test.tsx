import React from 'react';
import ReactDOM from 'react-dom';
import GameOver from '../GameOver';
import { render, cleanup  } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);

it('renders gameOver correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GameOver/>, div);
});

it('renders final score correctly', () => {
    const {getByTestId} = render(<GameOver score="1200"/>);
    expect(getByTestId('finalscore')).toHaveTextContent('1200');
});

it('matches snapshot', () => {
    const tree = TestRenderer.create(<GameOver score="1200"/>).toJSON();
    expect(tree).toMatchSnapshot();
});


