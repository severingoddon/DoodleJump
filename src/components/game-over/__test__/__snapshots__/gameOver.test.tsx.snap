// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="game-over-screen"
  style={Object {}}
>
  <div
    className="game-over"
  >
    Game over!
  </div>
  <div
    className="result-score"
    data-testid="finalscore"
  >
    Your score is: 
    1200
  </div>
</div>
`;
